### DISCUSSION TITLE
Fine tuning Demo RedBall

### DISCUSSION THREAD

**Discussion by justdiandes** [robotology/community#713] [Q&A]
Hi everyone!

So, I've been calibrating the `demoRedBall` and making the `stereoCalib` in order to make some studies related to the grasping and the vision itself. What I find interesting is that since the update from `v2022` to `v2024.02.0` in `icub-head`, the movement of the icub the `demoRedBall` with single camera is different in the sense that there's some behavior in the trajectory. For context on this I post the following videos:

https://github.com/user-attachments/assets/2d8f55ee-a964-40ab-b591-c477b9db7fba

https://github.com/user-attachments/assets/866c53d0-86d2-4295-81cc-613854292c6a

So, the icub has some problems in the movement itself, like if the robot had some issues with the joints or something related, but the only changes we've been doing is the software update. That's why I tried the calibration for the demoRedBall using the `calibOffset` module but didn't fix the problem. 

Additionally, there's the log outputs for the demoRedBall, yarprobotinterface and pf3dTracker here:
[demoredball.txt](https://github.com/user-attachments/files/16513278/demoredball.txt)
[pf3dtracker.txt](https://github.com/user-attachments/files/16513279/pf3dtracker.txt)
[yarprobotinterface.txt](https://github.com/user-attachments/files/16513280/yarprobotinterface.txt)

I wonder if there's any way to improve the performance in the robot movement and also if someone could guide in the stereo Red Ball module it will be really good.

Thanks in advance!

**Comment by pattacini** [comment]
Hi @justdiandes 

> the movement of the icub the demoRedBall with single camera is different in the sense that there's some behavior in the trajectory. 

What are exactly the strange behaviors you mentioned?
From the videos you posted, it appears to me that the robot behaves normally while performing the red-ball-demo, to the extent that it needs to be tuned.
